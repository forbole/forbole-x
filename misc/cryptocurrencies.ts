const cryptocurrencies = {
  // ATOM: { name: 'ATOM', image: '/static/images/cryptocurrencies/atom.png', coinType: 118 },
  DSM: {
    name: 'DSM',
    prefix: 'desmos',
    ecosystem: 'cosmos',
    image: '/static/images/cryptocurrencies/dsm.png',
    coinType: 852,
    graphqlHttpUrl: 'https://gql.morpheus.desmos.network/v1/graphql',
    graphqlWsUrl: 'wss://gql.morpheus.desmos.network/v1/graphql',
    blockExplorerBaseUrl: 'https://morpheus.desmos.network',
    defaultGasFee: {
      amount: [
        {
          amount: '2000',
          denom: 'udaric',
        },
      ],
      gas: {
        send: '100000',
        delegate: '200000',
        redelegate: '200000',
        claimRewards: '100000',
        undelegate: '200000',
      },
    },
  },
  // SOL: { name: 'SOL', image: '/static/images/cryptocurrencies/sol.png', coinType: 501 },
}

export default cryptocurrencies
