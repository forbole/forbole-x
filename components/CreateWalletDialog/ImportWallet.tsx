import {
  Button,
  DialogActions,
  DialogContent,
  Typography,
  Box,
  TextField,
  Avatar,
  ButtonBase,
  Grid,
  useTheme,
} from '@material-ui/core';
import useTranslation from 'next-translate/useTranslation';
import React from 'react';
import cryptocurrencies from '../../misc/cryptocurrencies';
import useStyles from './styles';

interface ImportWalletProps {
  onConfirm(name: string, cryptos: string[]): void;
}

const ImportWallet: React.FC<ImportWalletProps> = ({ onConfirm }) => {
  const { t } = useTranslation('common');
  const classes = useStyles();
  const theme = useTheme();
  const [name, setName] = React.useState('');
  const [selectedCryptos, setSelectedCryptos] = React.useState<string[]>([]);

  return (
    <form
      noValidate
      onSubmit={e => {
        e.preventDefault();
        onConfirm(name, selectedCryptos);
      }}>
      <DialogContent className={classes.dialogContent}>
        <Typography>{t('moniker')}</Typography>
        <TextField
          fullWidth
          variant="filled"
          placeholder={t('wallet name')}
          value={name}
          onChange={e => setName(e.target.value)}
          InputProps={{
            disableUnderline: true,
          }}
        />
        <Box mt={3} mb={28}>
          <Typography>{t('select currencies')}</Typography>
          <br />
          <Grid container spacing={2}>
            {Object.values(cryptocurrencies).map(c => {
              const isSelected = selectedCryptos.includes(c.name);
              return (
                <Grid key={c.name} item md={4}>
                  <ButtonBase
                    className={classes.cryptoButton}
                    style={isSelected ? { border: `1px solid ${theme.palette.text.primary}` } : {}}
                    onClick={() =>
                      setSelectedCryptos(cryptos =>
                        isSelected
                          ? cryptos.filter(crypto => crypto !== c.name)
                          : [...cryptos, c.name],
                      )
                    }>
                    <Box mr={2}>
                      <Avatar src={c.image} />
                    </Box>
                    <Typography color={isSelected ? 'textPrimary' : 'textSecondary'}>
                      {c.name}
                    </Typography>
                  </ButtonBase>
                </Grid>
              );
            })}
          </Grid>
        </Box>
      </DialogContent>
      <DialogActions>
        <Box flex={1} display="flex" flexDirection="column" mb={3}>
          <Button
            className={classes.button}
            variant="contained"
            color="primary"
            disabled={!name || selectedCryptos.length === 0}
            type="submit">
            {t('import')}
          </Button>
        </Box>
      </DialogActions>
    </form>
  );
};

export default ImportWallet;
