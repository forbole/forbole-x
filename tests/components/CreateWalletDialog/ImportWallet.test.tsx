import React from 'react';
import renderer from 'react-test-renderer';
import ImportWallet from '../../../components/CreateWalletDialog/ImportWallet';
import cryptocurrencies from '../../../misc/cryptocurrencies';

const onConfirm = jest.fn();

describe('component: CreateWalletDialog - ImportWallet', () => {
  it('renders correctly', () => {
    const component = renderer.create(<ImportWallet onConfirm={onConfirm} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with wallet name', () => {
    const component = renderer.create(<ImportWallet onConfirm={onConfirm} />);
    renderer.act(() => {
      component.root.findByType('input').props.onChange({
        target: {
          value: 'wallet',
        },
      });
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('calls onConfirm when button is clicked', () => {
    const component = renderer.create(<ImportWallet onConfirm={onConfirm} />);
    renderer.act(() => {
      component.root.findByType('input').props.onChange({
        target: {
          value: 'wallet',
        },
      });
    });
    // Click first crypto
    renderer.act(() => {
      component.root.findAllByType('button')[0].props.onClick();
    });
    // // Click second crypto
    // renderer.act(() => {
    //   component.root.findAllByType('button')[1].props.onClick()
    // })
    // // Click second crypto again
    // renderer.act(() => {
    //   component.root.findAllByType('button')[1].props.onClick()
    // })
    renderer.act(() => {
      component.root.findByType('form').props.onSubmit({ preventDefault: jest.fn() });
    });
    expect(onConfirm).toBeCalledWith('wallet', [Object.values(cryptocurrencies)[0].name]);
  });
});

afterEach(() => {
  jest.clearAllMocks();
});
