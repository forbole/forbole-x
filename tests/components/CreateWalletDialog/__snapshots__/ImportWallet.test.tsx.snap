// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: CreateWalletDialog - ImportWallet renders correctly 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      common:moniker
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="common:wallet name"
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        common:select currencies
      </p>
      <br />
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4"
        >
          <button
            className="MuiButtonBase-root HookGlobalStyles-cryptoButton-6"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={Object {}}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiBox-root MuiBox-root-10"
            >
              <div
                className="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  className="MuiAvatar-img"
                  src="/static/images/cryptocurrencies/dsm.png"
                />
              </div>
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              DSM
            </p>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-11"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:import
        </span>
      </button>
    </div>
  </div>
</form>
`;

exports[`component: CreateWalletDialog - ImportWallet renders correctly with wallet name 1`] = `
<form
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-7"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      common:moniker
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="common:wallet name"
          required={false}
          type="text"
          value="wallet"
        />
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-12"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        common:select currencies
      </p>
      <br />
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4"
        >
          <button
            className="MuiButtonBase-root HookGlobalStyles-cryptoButton-6"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            style={Object {}}
            tabIndex={0}
            type="button"
          >
            <div
              className="MuiBox-root MuiBox-root-13"
            >
              <div
                className="MuiAvatar-root MuiAvatar-circle"
              >
                <img
                  className="MuiAvatar-img"
                  src="/static/images/cryptocurrencies/dsm.png"
                />
              </div>
            </div>
            <p
              className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
            >
              DSM
            </p>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-14"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          common:import
        </span>
      </button>
    </div>
  </div>
</form>
`;
